platform/pulp/platform/models/task.py

```diff
@@ -0,0 +1,231 @@
+"""
+Django models related to the Tasking system
+"""
+from django.db import models
+
+from pulp.platform.models import Model
+from pulp.platform.fields import JSONField
+
+
+class ReservedResource(Model):
+    """
+    Resources that have been reserved
+
+    Fields:
+
+    :cvar resource: The name of the resource reserved for the task.
+    :type resource: models.TextField
+
+    Relations:
+
+    :cvar task: The task associated with this reservation
+    :type task: models.ForeignKey
+
+    :cvar worker: The worker associated with this reservation
+    :type worker: models.ForeignKey
+    """
+    resource = models.TextField()
+
+    task = models.ForeignKey("Task", unique=True)
```

Comment:
Changing this to a 1-1 field, per Django's suggestion:

`pulp_platform.ReservedResource.task: (fields.W342) Setting unique=True on a ForeignKey has the same effect as using a OneToOneField.
        HINT: ForeignKey(unique=True) is usually better served by a OneToOneField.`